utils/test/garden-test-utils.tsx

```diff
@@ -37,10 +37,23 @@ RtlProvider.propTypes = {
   children: PropTypes.node
 };
 
+const customRender = (ui: React.ReactElement, options?: any) => {
+  const results = render(ui, options);
+
+  return {
+    ...results,
+    getByGardenId: (id: string, container?: HTMLElement) => {
+      return (container || results.container).querySelector(`[data-garden-id="${id}"]`);
+    },
+    getAllByGardenId: (id: string, container?: HTMLElement) => {
+      return (container || results.container).querySelectorAll(`[data-garden-id="${id}"]`);
+    }
```

Comment:
The two selectors were helpful in making the tests easier to work with when selecting Garden elements - it is by no means necessary nor is it the appropriate PR to include this, happy to remove from this PR.

Reply:
The preferred mechanism is to add a `data-test-id` which gets stripped during build. Grep for details.

Reply:
Another alternative is to query the element in an [accessible way](https://testing-library.com/docs/queries/about#priority). In areas where `getByGardenId` is used, the `getByText` could be used instead. If a [label](https://html.spec.whatwg.org/multipage/forms.html#the-label-element) was rendered, then the `getByLabelText` query could be used. For example,

```jsx
it('renders with badge and with a provided status label', () => {
  const { getByText } = render(
    <Avatar badge="2" statusLabel="two notifications">
      <img alt="lorem ipsum" />
    </Avatar>
  );

  const statusText = getByText("two notifications");

  expect(statusText).toBeInTheDocument();
});
```

There's also a bunch other accessible query functions like `getByRole`, `getByAltText`, and `getByTitle` ♿ .

Reply:
Yeah, even better ☝️. Thanks @hzhu.